### PR TITLE
Added Courses API

### PR DESCRIPTION
**Pull request by MozdolevskyiMaksym**
**Get list of courses** `/api/courses`
_Response body:_ 
```
[
 {
  id: int,
  name: string
 },
 ...
]
```
----------------------------------
**Post a course** `/api/courses`
_Request body:_ 
```
{
 name: string
}
```
_Response body:_
```
{
 name: string
}
```
------------------------------------
**Put a course** `api/courses/[id]`
_Request body:_
```
{
 name: string
}
```
_Response body:_
```
{
 id: int,
 name: string
}
```
------------------------------------------
**Delete a course** `/api/courses/[id]`

### CODE REVIEW

**Review by Anastasiia0999 — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -1,12 +1,95 @@
...
+}))
+
 app.get('/api', ((req, res) => {
   res.send('Hello from WHAT mock server!');
 }));
 
+app.get('/api/courses', ((req, res) => {
+  fs.readFile('./mocks/courses.json', ((err, data) => {
+    if (err) {
+      req.status(500).send('Server error occured');
```
Comment (rearchontis) [inline index.js]: How do you know that you should send 500 response status?
Maybe you should `throw err`?

File: index.js [hunk 2]
```diff
@@ -1,12 +1,95 @@
...
+      const responseData = JSON.parse(data);
+      res.send(responseData);
+    }
+  }));
+}));
+
+app.post('/api/courses', ((req, res) => {
+  fs.readFile('./mocks/courses.json', ((err, data) => {
+    if (err) {
+      req.status(500).send('Server error occured');
```
Comment (rearchontis) [inline index.js]: Just use `throw Error(err)` or something like that

File: index.js [hunk 3]
```diff
@@ -1,12 +1,95 @@
...
+app.put('/api/courses/:id', ((req, res) => {
+  const courseId = Number(req.params.id);
+  const {name} = req.body;
+  const editedCourse = {
+    id: 50,
+    name,
+  };
+  fs.readFile('./mocks/courses.json', ((err, data) => {
+    if(err) {
+      res.status(500).send('Server error occured');
```
Comment (rearchontis) [inline index.js]: Just remind about throwing errors

File: index.js [hunk 4]
```diff
@@ -1,12 +1,95 @@
...
+    }
+
+    const courses = JSON.parse(data.toString());
+    const course = courses.find((course) => course.id === courseId);
+    if (course) {
+      res.send(`course with id ${courseId} is edited`);
+    } else {
+      res.status(403).send(`There is no course with id ${courseId}`);
```
Comment (rearchontis) [inline index.js]: Undefined behavior here. I don't have the right to make put requests or I am trying to fetch a course that doesn't exist?

403 status tells you that you don't have enough rights to fetch or change something.
Read about [403 response status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403).

But the message you send is not assignable to 403 status.

File: index.js [hunk 5]
```diff
@@ -1,12 +1,95 @@
...
+    }
+  }));
+}));
+
+app.delete('/api/courses/:id', ((req, res) => {
+  const id = Number(req.params.id);
+
+  fs.readFile('./mocks/courses.json', ((err, data) => {
+    if(err) {
+      res.status(500).send('Server error occured');
```
Comment (rearchontis) [inline index.js]: Change this line

File: index.js [hunk 6]
```diff
@@ -1,12 +1,95 @@
...
+    }
+    const courses = JSON.parse(data.toString());
+    const course = courses.find((course) => course.id === id);
+
+    if(course) {
+      res.send(`Course with id ${id} is deleted`);
+    } else {
+      res.status(403).send(`There is no course with id ${id}`);
```
Comment (rearchontis) [inline index.js]: Read the comment above with the same line at the put request handler

**Review by rearchontis — APPROVED** [review]

**Review by so2niko — APPROVED** [review]

**Pull request merged**